### ISSUE TITLE
Design Proposal - Mechanical Limbs Electric Boogaloo

### ISSUE THREAD

**Issue by N3X15** [d3athrow/vgstation13#24908]
This is a design proposal, not set in stone, here for feedback and so I don't forget what I'm doing.

## TL;DR
* Current organ code is an unfinished, unbalanced mess, so this needed to be done anyway.
* Add synthetic limbs and organs with upgrade paths
  * Each organ has [0,n) upgrade slots
  * Organs can be customized with skins that don't affect functionality
* Upgrades are things like
  * Filters for air/blood
  * Weapons
  * Tracking beacons
  * Communications (ears only)
  * Armor plating
  * HUD modules
  * pAIs (head, take over job of borer)
  * Function boosters
  * Storage
  * SPS
  * etc
* Remove fatal EMP effects
  * Add lesser effects to some parts.
* Remove need of medbay/robotics interaction by adding autodocs
  * Placed in public hallways like PDA app vendors
  * Sell synthetic limbs and organs
  * Can repair damaged synthetic limbs, for a price
  * Discounted price if the limb is already missing (less surgical shit)
  * Sell several different designs/skins
  * Offer upgrade modules for existing organs
  * Will replace organic organs
  * Sells contraband upgrades (weapons, etc) for telecrystals or money when emagged

## Organs In General
* Heart - Affects movespeed
* Liver - Filters reagents
* Lungs - Gas exchange and filtering
* Arms - Attacks, interactions, weapons
* Torso - Armor
* Legs - Movespeed, balance (slip)
* Eyes - Sight, HUDs
* Ears (NEW) - Hearing range, speech filtering, comms

### Discount Dan
**Cost:** Almost free
**Designs:** Redneck engineering (cobbled together wooden/rusty shit)
**Skins:** None
**Organ Inventory:**
* Peg limbs (1/2 movespeed)
* Accordion lungs (Poor gas exchange, causes you to get winded when running)
  * Can be played
* Turkey baster heart (1/2 movespeed)
* Glass eyes (no eyesight)

**Upgrades:** None.

### DOGG
(A play on CAT, AKA Caterpillar)
**Cost:** Cheap
**Designs:** Utilitarian, industrial. No curved surfaces (except for things that need them, like ball joints).  Exposed hydraulics, etc.
**Upgrades:**
  * Some specialized industrial shit thrown in like mining drills and power lifters (expensive)

**Implant Inventory:**
  * Chempacks:
    * RadAway
    * Kelotane (for burns)
  * GPS
  * Surge Protector 1000MJ - Sacrificial, protects from one (1) EMP.
  * Surge Protector 5000MJ - Sacrificial, protects from five (5) EMPs.

**Organ Inventory:**
  * OP1 - Basic cylindrical eye with fisheye lens. Looks inhuman. 
    * Upgrade: Meson (requires meson goggles for CCD)
    * Upgrade: Thermal (requires thermal goggles for CCD)
    * Upgrade: Cat Eye (Requires tajaran eyeballs)
  * STRDY legs - Heavy duty thicc mecha legs. Slow but can be upgraded with anchoring spikes (only work on asteroid/snow) or magnetic anchors (only work on metal surfaces). Both behave like magboots and slow movement when active. 
    * Upgrade: Anchoring Spikes
    * Upgrade: Magnetic Anchors
    * Skins: Asteroid Brown, Safety Yellow, Black, and Gray
  * HLP1 legs: Utilitarian humanoid plantigrade legs, look like old early 21st century prosthetics, with exposed parts.
    * Colors: Asteroid Brown, Cargo Two-Tone, Deere, Security Black and Red, Medbay White/Blue, Research White/Purple, Black, Grey, Clown Wrap.
    * Upgrade: Enhanced Gyros - Drops chance of slipping or being blown over.
    * Upgrade: Enhanced Actuators - Slightly faster movespeed.
  * HLD1 legs: Same as HLP1, but digitigrade (for vox or whatever). Warns the user on the screen.  Can be used by anyone who want backwards knees.
  * HLP2/HLD2: Built with the same reliable DOGG powertrain and stabilization kit as the P1s, but with everything rerouted to make way for a plastic exoskeleton that produces a rough humanoid outline. Still cheap and utilititarian, but causes fewer weird looks in the bar. (HLP2 = Plantigrade, HLD2 = Digitigrade)
    * Colors: Same as HLP1
  * MaxPower Arms - Beefy industrial arms built for handling heavy loads. You can pick up a crate/ore box with both hands with these. Fairly expensive.
    * Skins: Asteroid Brown, Safety Yellow, Black, and Gray
    * Upgrade: Power Loading - Makes picking up crates faster
    * Upgrade: Worklights - Shoulder mounted LED floodlights
  * HA1 Arms - Can move in the same way as a human arm, but with no exterior exoskeleton to hide all the hydraulics, pneumatics, and wiring. 
    * Colors: Asteroid Brown, Cargo Two-Tone, Deere, Security Black and Red, Medbay White/Blue, Research White/Purple, Black, Grey, Clown Wrap.
  * HA2 Arms - Like the HLP2s, a baseline model of arms, but with an exterior exoskeleton for some level of style. 
    * Colors: Asteroid Brown, Cargo Two-Tone, Deere, Security Black and Red, Medbay White/Blue, Research White/Purple, Black, Grey, Clown Wrap.
  * HT1 Core - A bare cybernetic torso with exposed parts. Includes built-in heart, lungs, and liver, all certified for baseline humanoid survival.
    * Upgrade: Emergency Oxygen - Slowly refills when breathing oxygen atmosphere
    * Upgrade: Emergency Plasma - Slowly refills when breathing plasma atmosphere
    * Upgrade: Emergency Nitrogen - Slowly refills when breathing nitrogen atmosphere 
    * Colors: Asteroid Brown, Cargo Two-Tone, Deere, Security Black and Red, Medbay White/Blue, Research White/Purple, Black, Grey, Clown Wrap.
  * HT2 Core - Same as the HT2, but with an exoskeleton.
  * HH1 - Vaguely human head with two basic eyes and ears. Provides only the necessary parts needed for basic function (two eyes, two ears, a mouth) bolted onto a barebones frame.
    * Upgrade: Integrated Comms (Crypto keys sold seperately)
    * Upgrade: pAI Slot (pAI sold seperately)
  * HH2 - HH1 with an exoskeleton. Eyes and Ears are still basic.

### Boston Bionics
**Cost:** More expensive
**Design:** Look natural or near-natural
**Implant Inventory:**
  * Cochlear Comms - Internal comms for meatbags
  * Hearing Aid - Hear more distant things. (TODO: determine if this is even possible with saycode)
  * Dental Implants - Quick way of replacing your teeth after someone's beaten your face in with a toolbox
  * Cat Ears - Cosmetic

**Organ Inventory**
  * Prosthetic Series  - Human analogues.  Automatically detect skin tone. Look natural, beyond a seam or port here and there.
  * Prosthetic Eyes - Natural-looking eyes
  * Prosthetic+ Series - Upgradable human analogues
    * Torso Upgrade: Cyberpunk - Togglable glowy lines.

### Bioelectic
**Cost:** High-End
**Design:** Sleek, some skins with almost alien appearance
**Implants:**
  * Medical Monitoring
  * Borer Hatchery (requires plasma, organic torso)
  * Retinal MedHUD (requires organic eye)
  * Adaptive Contact Lens - Eliminates welder damage to eyes (requires organic eye and welding goggles)

**Organ Inventory:**
  * SpEye Camera - Normal looking eye, but can take pictures.
  * Adaptive Eye - Eliminates welding damage (and flash; Requires welding goggles)
  * Multispectal Eyes - Allows you to select which spectrum or particle you see. User ends up looking like a Mechanicus Adept in terms of lenses-to-face quotient.
     * Upgrade: Meson - Requires meson goggles
     * Upgrade: Thermal - Requires thermal goggles
     * Upgrade: T-Ray - Requires T-ray scanner
     * Upgrade: MedHUD - Requires MedHUD
     * Upgrade: SecHUD - Requires SecHUD
     * Colors: Green, red, cyan, purple, orange, Grey visor, Black Visor, White Visor, Red Visor
  * Squat Head - Reduces the head to little more than a hill on the shoulders that your eyes pop out of.  
  * GoRILA Arms - Longer arms capable of reaching one tile longer than most arms. They drag on the floor and require larger shoulders, however, resulting in a significant movement speed impact.
    * Skins: Bioelectric sleek, Bare
    * Colors: Green, Red, Cyan, Purple, Orange, White, Grey.
  * Medical R&D Limbs - Easy to clean arms made largely out of thick sheets of ceramic.
    * Upgrade: Self Clean - Arms can drench themselves in spacecleaner, which refills slowly.
    * Upgrade: Multitool - Embeds a holotool into the dominant hand's fingertips, summonable with an ability button.  Requires a holotool.
    * Upgrade: Hypospray - Embeds a hypospray into the dominant hand's fingertips, summonable with an ability button. Requires a hypospray.

### Northrippe Martin
**Cost:** High-End, billed to security/station account.
**Restrictions:** Military and security only. Requires sec access or emag, vendor located in armory.
**Implants:**
  * SPS - Requires SPS.
  * Retinal SecHUD - Requires SecHUD

**Universal Upgrades:** 
  * 9mm Wrist Cannon - Summonable with ability. 6 shot 9mm pistol added to wrist.  Requires free hand. Low accuracy. Basically a shit glock. Requires glock.
    * Upgrade: Silencer
  * Wrist Taser - Same as above, except it's a taser. Requires taser.
    * Upgrade: High-pressure gas cylinder - Doubles movement speed of taser bolt.
    * Upgrade: High-voltage Capacitors - Increases stun time
  * Extendable baton - Arm-mounted stun baton. Requires stun baton.
  * Riot shield mount
  * Holster

**Organs:**
  * Summonable Jensen Glasses
  * SWAT Torso/Arms/Legs - Like the DOGG H#1s, but plated in armor. Equivalent to the existing vests.
    * Upgrade: High Speed Chase Module - Replaces feet with high-torque motors and wheels, like the EMW GFST.

### EMW
  * High-End
  * Speed/Racing focus
  * Sleek, aerodynamic appearance
  * GFST Legs - MUST be used with upgraded heart (max movespeed, but cannot turn while moving, brute when you slip/run into something)
  * Turbopump heart

## Autodoc
* Basically like a cylindrical phonebooth
  * Actually just a fancy locker with a screen that pops up when closed
  * Occupied light/sign
* Each can have its own selection of stuff
  * Armory has special weapons-only vendor
* Performs limb replacement within 10s (or faster when mechanics upgrade it)
  * Makes sawing, squelching, screaming sounds when in use
* Charges paid to medbay account
* Like vending machine, provides stuff from one company only(?), unique design for that company
* Detects existing synth limbs, displays upgrade options for them.
* If you are inside without interaction for 30s, will give you a 10 second countdown, then open the door and forcefully kick you out (literally stuns you and throws you).
* Wires:
  * Employee discount
  * Stepper resolution (causes random limbs to get sawn off when used)
  * Shock
  * Power
  * Door
  * Anti-hobo spray

## Organ Printers
* Print out cheap humanoid-equivalent parts with a few upgrade slots.
* Remove natural organ printing.  Fuck you, use a monkey or a dead assistant.

# FAQ
<dl>
<dt>Whaddabout FBPs?</dt>
<dd>Full-Body Prosthetics as a *concept* will be possible simply by replacing all of your squishy human bits with prosthetics. As for Polaris sprites, I need to talk to them so we don't cause too much drama.</dd>
<dt>Whaddabout IPCs?</dt>
<dd>Later, maybe.  Right now, these are just humans/vox/whatevers with cybernetic parts.  They'll still have to consider consuming oxygen/nitrogen/plasma/CO2 for keeping their brains and organics going.</dd>
</dl>

**Comment by MisterStaub** [comment]
Something i've always wanted from spess but never really got.
Do you plan on locking augmentations to autodocs, or are you considering manual augmentation via surgery as well?

**Comment by utilitas-** [comment]
how about making skin an organ as well? then you could make true FBPs by replacing your skin too
skin would allow to code in chemistry/syringe resistance in a non-shitty way too

**Comment by N3X15** [comment]
I'm making each limb have a skin that you can change like you'd change the paint on a car.  Having a skin organ would interfere with that.

**Comment by utilitas-** [comment]
yes but what about _subdermal plate armor_?

**Comment by N3X15** [comment]
Yep, as an upgrade.

**Comment by utilitas-** [comment]
finally, I can make my dreams come true

> Jane-Luc Price lets a small grin grow on her face as she slams her head forward, cracking into Ashleys, the impact seeming not to effect her as she taps her skull "Dermal armor." with a grin, she'd slide her hand into Ashleys panties

**Comment by Shadowmech88** [comment]
>Full-Body Prosthetics as a *concept* will be possible simply by replacing all of your squishy human bits with prosthetics.

Will robotic torsos be a thing? Even if you replace all your organs, the torso/groin itself would still be fleshy.

Also, assuming you're going to be largely rewriting organ and limb code, and given how much of human health the organs and limbs make up, how close would this get us to something like brainmed?

**Comment by N3X15** [comment]
> Will robotic torsos be a thing? Even if you replace all your organs, the torso/groin itself would still be fleshy.

Yep.

> Also, assuming you're going to be largely rewriting organ and limb code, and given how much of human health the organs and limbs make up, how close would this get us to something like brainmed?

I'm going to be moving as much as I can into organ code, much like I did when I made the first iteration of this with lungs.  

> Brainmed

wat

**Comment by Probe1** [comment]
I'm surprisingly fine with autodocs.  Normally I'm constantly complaining that we need to force player interaction through game design because this is a multiplayer game and it thrives on player interaction.  But we've got half a decade of experience showing that elective surgeries just don't happen and people go to Medbay for treatment, not improvement.

**Comment by N3X15** [comment]
Only half a decade?

**Comment by mentgah** [comment]
Give roboticists something to do.

**Comment by N3X15** [comment]
@mentgah I just can't see roboticists getting involved, due to a combination of the sheer demand and having their own projects.  They'd have to be doing limb surgery the entire round, or pumping out parts.

**Comment by Zth--** [comment]
![suicide booth](http://horrorfuel.com/wp-content/uploads/2018/04/suicide_booth_by_r_w_shilling.jpg)
emagged autodoc removes all your limbs when?

**Comment by Caboozles** [comment]
I dislike the idea of buffing these augs through the roof while simultaneously nerfing how impactful EMPs are. If you want to increase their effectiveness and not make EMP’d hearts instakill that’s fine. But they should have some pretty major drawbacks if EMP’d so we don’t get into a snowflake war, especially since some of this stuff sounds actively incredible, and invites powergaming out the wazoo. This needs a real limitation beyond money, because people will always find a way around that.

**Comment by mentgah** [comment]
fair enough

**Comment by utilitas-** [comment]
Autodocs should still be something that needs to be built by RnD/ordered, though, like the rigsuit modification station
maybe give robotics a frame
it's either you put in an autodoc that does all the operations or make organs "hot swappable", where you have to only do the organ surgery once and then be able to replace your organs with a relatively quick swap at an autodoc station

**Comment by utilitas-** [comment]
Oh and, while we're at it
[spoiler]vorestation[/spoiler] has a kinda neat thing: Nanite Implant Framework. Inserted into the brain with a surgery, it's a chip that allows inter-round persistence. We don't really give a shit about that, but it also allows you to install departmental HUDs with specific chips, and an illegal chip allows you to install laws much like with silicons.
Maybe allow the brain, be it organic or whatever, to be upgraded as well? Something something brain uploading/transferral/backup, which then can take advantage from hard-wired departmental HUDs and other neat shit.

**Comment by Blithering** [comment]
I think initial manual surgery to make you autodoc compatible, then from then on hotswap via phonebooth would be a good way to go about it, that way you don't have to nag a doctor to do every single surgery, but you don't remove them from the equation altogether. Would also be up for some spriting if polaris is a no go.

**Comment by N3X15** [comment]
Again, I don't see neither doctors nor robotics handling the dozens of surgeries per round necessary to keep up with demand.  I can of course leave in manual limb attachment, but I just don't see them being used.

**Comment by Blithering** [comment]
Fair enough, we can try it and see how it goes. Also for the Northrippe Martin, can I suggest a carbon fiber/adam jensen skin style? Tacticool enough to be popular for sec, and not too human as to step on Boston Bionics' toes.

**Comment by N3X15** [comment]
Yeah, was going to go with various camos, black, USA, and skin shades.

**Comment by Zth--** [comment]
What about a clown horn as an accordion lung? It'd honk every time you breathe (or maybe not every time to avoid it getting old very fast)

**Comment by SageOfLight** [comment]
Which option gives me the most weeaboo looking prosthetics.

Also how will this effect players who start with bionics? Will they get a choice for aesthetic/exoskeletons? I want my gundam arms.

**Comment by Shadowmech88** [comment]
@N3X15 
Brainmed refers to the idea of doing away with health as a number, and instead basing a spessman's life or death solely on the amount of brain damage they take, in an effort to be more true to life. Wounds are dangerous not because they remove health, but because they cause blood loss which can cause oxygen deprivation which causes brain damage, that sort of thing.

**Comment by juanototo2** [comment]
Also for Northrippe's lethals try to do it in lockboxes or add a way to open them. I don't want to see YET MORE shit relying on getting the spare.

**Comment by N3X15** [comment]
Northrippe's shit would only be available in a special Northrippe autodoc in the armory.  You could order the parts from cargo and build it like a vending machine, but the installation surgery would require sec access/emag.  You could also make the organs manually in robotics, but yeah, lockboxes.

**Comment by tayachiya** [comment]
Would it be possible to have some option that if you spawn with metal limbs your limbs don't grow back if you get cloned?

**Comment by Blithering** [comment]
I think cloning should always reset you to all natural limbs and organs, partially because it makes more sense, and partially because if we add this, then being reset to "factory zero" would be a major negative to the kill and clone medbay mindset.

**Comment by utilitas-** [comment]
\>getting cloned means you lose all your fancy (albeit potentially fucked up) robolimbs and roboorgans
\>becomes a genuine option instead of an alternative to the occasional choose your own adventure of mending twenty fractures and internal bleedings
yes please

**Comment by N3X15** [comment]
Agreed with getting natural limbs on cloning.  Is that not how it's currently done?

**Comment by strawberrytan** [comment]
I like my surgery autism even if volunteers are hard to come by

**Comment by N3X15** [comment]
> I like my surgery autism even if volunteers are hard to come by

Monkeys!  All the fun, zero ethical risk.

**Comment by utilitas-** [comment]
strawberrytan prefers their victims to scream bloody murder, not stare catatonic.
